tests/integration/models/applicationInvite.spec.ts

```diff
@@ -0,0 +1,132 @@
+					...args: any[]
+				) => any)(this.application.id, function (
+					_err: Error,
+					applicationInvite: BalenaSdk.ApplicationInvite[],
+				) {
+					expect(applicationInvite).to.deep.equal([]);
+					return done();
+				});
+			});
+		});
+
+		describe('balena.models.application.invite.create()', function () {
+			it('should create and return an application invite', async function () {
+				const response = await balena.models.application.invite.create(
+					this.application.id,
+					{
+						collaborator: TEST_EMAIL,
+						roleName: TEST_ROLE,
+						message: TEST_MESSAGE,
+					},
+				);
+				expect(response).to.have.property('id');
+				const invites = await balena.models.application.invite.getAllByApplication(
+					this.application.id,
+					{
+						$expand: {
+							is_invited_to__application: { $select: ['id'] },
+							invitee: { $select: ['email'] },
+							application_membership_role: { $select: ['name'] },
+						},
+						$select: ['message'],
+					},
+				);
+				expect(invites).to.have.length(1);
+				expect(invites[0].is_invited_to__application[0].id).to.equal(
+					this.application.id,
+				);
+				expect(invites[0].application_membership_role[0].name).to.equal(
+					TEST_ROLE,
+				);
+				expect(invites[0].message).to.equal(TEST_MESSAGE);
+			});
+		});
+	});
+
+	describe('given a single application invite [contained scenario]', function () {
+		givenAnApplication(before);
+		before(async function () {
+			const applicationInvite = await balena.models.application.invite.create(
+				this.application.id,
+				{
+					collaborator: TEST_EMAIL,
+					roleName: TEST_ROLE,
+					message: TEST_MESSAGE,
+				},
+			);
+			return (this.applicationInvite = applicationInvite);
+		});
+
+		describe('balena.models.application.invite.getAllApplication()', () => {
+			it('should become the list of application invites', async function () {
+				const applicationInvites = await balena.models.application.invite.getAllByApplication(
+					this.application.id,
+				);
+				expect(applicationInvites).to.have.length(1);
+			});
+
+			it('should support arbitrary pinejs options', async function () {
+				const applicationInvites = await balena.models.application.invite.getAllByApplication(
+					this.application.id,
+					{
+						$expand: { invitee: { $select: ['email'] } },
+					},
+				);
+				expect(applicationInvites[0].invitee[0].email).to.equal(TEST_EMAIL);
+			});
+		});
+
+		describe('balena.models.application.invite.get()', function () {
+			it('should be able to get an invite', async function () {
+				const applicationInvite = await balena.models.application.invite.get(
+					this.application.invite.id,
+				);
+				expect(applicationInvite.message).to.equal(TEST_MESSAGE);
+			});
+
+			it('should be rejected if the applicationInvite id is invalid', function () {
+				const promise = balena.models.application.invite.get(99999999999);
+				return expect(promise).to.be.rejectedWith('Request error');
```

Comment:
It would be great to also check the status code w/ something like
```
				return expect(promise).to.be.rejected.then(function (error) {
					expect(error).to.have.property('code', 'BalenaRequestError');
					expect(error).to.have.property('statusCode', 404); // For example. I don't know what this really returns.
					expect(error)
						.to.have.property('message')
						.that.contains('Request error');
				});
```